site/README.md

```diff
@@ -0,0 +1,165 @@
+# NGINX Gateway Fabric Documentation
+
+This directory contains all of the user documentation for NGINX Gateway Fabric, as well as the requirements for building and publishing the documentation.
+
+Documentation is written in Markdown, built using [Hugo](https://gohugo.io) with [nginx-hugo-theme](https://github.com/nginxinc/nginx-hugo-theme), then deployed with [Netlify](https://www.netlify.com/).
+
+## Setup
+
+Hugo is the only requirement for building documentation, but the repository's integration tooling uses markdownlint-cli.
+
+> **Note**: We currently use [Hugo v0.115.3](https://github.com/gohugoio/hugo/releases/tag/v0.115.3) in production.
+
+Although not a strict requirement, markdown-link-check is also used in documentation development.
+
+If you have [Docker](https://www.docker.com/get-started/) installed, there are fallbacks for all in the [Makefile](Makefile), meaning you do need to install them.
+
+- [Installing Hugo](https://gohugo.io/getting-started/installing/)
+- [Installing markdownlint-cli](https://github.com/igorshubovych/markdownlint-cli?tab=readme-ov-file#installation)
+- [Installing markdown-link-check](https://github.com/tcort/markdown-link-check?tab=readme-ov-file#installation).
+
+The configuration files are as follows:
+
+- *Hugo*: `config/default/config.toml`
+- *markdownlint-cli*: `mdlint_conf.json`
+- *markdown-link-check* `md-linkcheck-config.json`
+
+## Repository guidelines
+
+Documentation follows the conventions of the regular codebase: use the following guides.
+
+- [Pull Request Guidelines](../docs/developer/pull-request.md)
+- [Branching and Workflow](../docs/developerr/branching-and-workflow.md)
+- [Release Process](../docs/developer/developer/release-process.md)
+
+To work on documentation, create a feature branch in a forked repository then target `main` with your pull requests, which is the default repository branch.
+
+The documentation is published from the latest public release branch. If your changes require immediate publication, create a pull request to cherry-pick changes from `main` to the public release branch.
+
+## Developing documentation locally
+
+To build the documentation locally, run the `make` command inside this `/site/` directory:
+
+```text
+make docs            - Builds the documentation set with the output as the '/public' directory
+make clean           - Removes the local '/public/' directory
+make watch           - Starts a local Hugo server for live previews
+make watch-drafts    - Starts a local Hugo server for live previews, including documentation marked with 'draft: true'
+make link-check      - Check for any broken links in the documentation
+make lint-markdown   - Runs markdownlint to identify possible markdown formatting issues
```

Comment:
Do you want to include all the Makefile targets in this list? There are a few that are not included, such as `deploy-preview`, `build`, `all-dev`, etc.